src/ucp/core/ucp_worker.c

```diff
@@ -1366,11 +1366,10 @@ ucs_status_t ucp_worker_create(ucp_context_h context,
 
     if (params->field_mask & UCP_WORKER_PARAM_FIELD_THREAD_MODE) {
 #if !ENABLE_MT
-        if (params->thread_mode == UCS_THREAD_MODE_MULTI) {
-            return UCS_ERR_INVALID_PARAM;
-        }
-#endif
+        thread_mode = UCS_THREAD_MODE_SINGLE;
```

Comment:
shell we issue some warring is we simply ignore/overwrite thread_mode ?

Reply:
don't think so - the behavior is same as for MPI_Init_thread - requested thread level may be different from supported thread level

Reply:
@hoopoepg let's add API documentation to ucp.h worker_create and ucp_init that the actual thread mode may differe than requested, and user should use query to find the actual thread mode.

Reply:
we can also add debug print (but not warning/error)

Reply:
@yosefe docs are updated

Reply:
thanks! do we need to update the doc of ucp_init() as well?

Reply:
why not... will add another commit

Reply:
@yosefe updated desc for ucp_param_t